### PR TITLE
Tests for bitwise operations AND, OR and XOR

### PR DESCRIPTION
**Pull request by avilches**

### CODE REVIEW

File: core/src/test/java/org/bitcoinj/script/ScriptTest.java [hunk 1]
```diff
@@ -226,32 +226,65 @@ public void createAndUpdateEmptyInputScript() throws Exception {
...
                 long val = Long.parseLong(w);
                 if (val >= -1 && val <= 16)
-                    out.write(Script.encodeToOpN((int)val));
+                    out.write(Script.encodeToOpN((int) val));
                 else
                     Script.writeBytes(out, Utils.reverseBytes(Utils.encodeMPI(BigInteger.valueOf(val), false)));
             } else if (w.matches("^0x[0-9a-fA-F]*$")) {
                 // Raw hex data, inserted NOT pushed onto stack:
                 out.write(HEX.decode(w.substring(2).toLowerCase()));
+            } else if (w.matches("^bytes\\(0x[0-9a-fA-F]*\\)$")) {
```
Comment (avilches) [inline core/src/test/java/org/bitcoinj/script/ScriptTest.java:285]: I had to create a new way to insert bytes into the stack. So now you can do "bytes(0xF0) bytes(0x0F) OR" and you will get 0xFF in the stack. If you don't like the "bytes(0x.....)" approach, I can change it

File: core/src/test/java/org/bitcoinj/script/ScriptTest.java [hunk 2]
```diff
@@ -226,32 +226,65 @@ public void createAndUpdateEmptyInputScript() throws Exception {
...
 
         LinkedList<byte[]> stack = new LinkedList<byte[]>();
         Script.executeScript(tx, 0, script, stack, Script.ALL_VERIFY_FLAGS);
         assertEquals("OP_0 push length", 0, stack.get(0).length);
     }
 
+    static final int MAX_BITWISE_RANDOM_TESTS = 2000;
+
+    @Test
+    public void testBitwiseOneByteLength() throws IOException {
```
Comment (avilches) [inline core/src/test/java/org/bitcoinj/script/ScriptTest.java:240]: This test only checks one-single-byte operations in the stack, but bitwise operations with multiple bytes in the stack can be created also.

**Comment by HashEngineering** [comment]
Why was this closed?

**Comment by avilches** [comment]
Hi. I sent this PR to https://github.com/shadders/bitcoinj-cash-dev/pull/1 instead. Then @shadders will PR al the op-codes and tests again all together to this project.

**Pull request closed without merging**